packages/react/src/utils/noop.ts

```diff
@@ -1 +1 @@
-export const noop = (): void => {}
+export const noop = (): void => void 0
```

Comment:
below is also alternative.

```suggestion
export const noop = (): void => undefined
```

I want to get benchmark for these to choose best one. lint is not important thing in my opinion

1. export const noop = (): void => void 0
2. export const noop = (): void => undefined
3. export const noop = (): void => {}
1. export function noop(): void {return void 0}
2. export function noop(): void {return undefined}
3. export function noop(): void {}

Reply:
> below is also alternative.
> 
> I want to get benchmark for these to choose best one. lint is not important thing in my opinion
> 
> 1. export const noop = (): void => void 0
> 2. export const noop = (): void => undefined
> 3. export const noop = (): void => {}
> 4. export function noop(): void {return void 0}
> 5. export function noop(): void {return undefined}
> 6. export function noop(): void {}

@manudeli 

```ts
import Benchmark from 'benchmark';

const noopArrowVoid0 = (): void => void 0;
const noopArrowUndefined = (): void => undefined;
const noopArrowEmpty = (): void => {};
function noopFnVoid0(): void {
  return void 0;
}
function noopFnUndefined(): void {
  return undefined;
}
function noopFnEmpty(): void {}

const suite = new Benchmark.Suite();

suite
  .add('arrow => void 0', () => {
    noopArrowVoid0();
  })
  .add('arrow => undefined', () => {
    noopArrowUndefined();
  })
  .add('arrow => {}', () => {
    noopArrowEmpty();
  })
  .add('function return void 0', () => {
    noopFnVoid0();
  })
  .add('function return undefined', () => {
    noopFnUndefined();
  })
  .add('function {}', () => {
    noopFnEmpty();
  })
  .on('cycle', (event: Benchmark.Event) => {
    console.log(String(event.target));
  })
  .on('complete', function (this: Benchmark.Suite) {
    console.log('Fastest is', this.filter('fastest').map('name'));
  })
  .run({ async: true });
```
Thanks for the suggestion. I ran a benchmark to compare the different noop implementations using [benchmark](https://www.npmjs.com/package/benchmark), and here are the results:

```bash
// Run 1
arrow => void 0               x 211,230,560 ops/sec ±6.20%
arrow => undefined            x 221,422,561 ops/sec ±5.72%
arrow => {}                   x 233,960,019 ops/sec ±4.69%
function return void 0        x 226,863,806 ops/sec ±5.20%
function return undefined     x 215,939,870 ops/sec ±5.24%
function {}                   x 220,901,111 ops/sec ±5.13%

Fastest group: 
[ 'arrow => {}', 'function return void 0', 'function {}', 'arrow => undefined' ]

// Run 2
arrow => void 0               x 210,357,565 ops/sec ±5.45%
arrow => undefined            x 221,853,911 ops/sec ±4.68%
arrow => {}                   x 231,361,189 ops/sec ±3.91%
function return void 0        x 228,022,530 ops/sec ±4.76%
function return undefined     x 224,487,843 ops/sec ±4.49%
function {}                   x 214,505,613 ops/sec ±5.34%

Fastest is [ 'arrow => {}', 'function return void 0', 'function return undefined', 'arrow => undefined' ]

// Run 3
arrow => void 0               x 223,469,257 ops/sec ±4.82%
arrow => undefined            x 226,982,491 ops/sec ±3.80%
arrow => {}                   x 220,358,115 ops/sec ±5.12%
function return void 0        x 218,073,613 ops/sec ±5.31%
function return undefined     x 215,593,699 ops/sec ±5.26%
function {}                   x 232,004,784 ops/sec ±3.91%

Fastest is [ 'function {}', 'arrow => undefined', 'arrow => void 0', 'arrow => {}', 'function return void 0', 'function return undefined' ]

// Run 4
arrow => void 0               x 221,303,342 ops/sec ±4.72%
arrow => undefined            x 216,896,414 ops/sec ±4.98%
arrow => {}                   x 225,387,219 ops/sec ±4.88%
function return void 0        x 223,999,827 ops/sec ±4.70%
function return undefined     x 218,475,438 ops/sec ±4.50%
function {}                   x 212,648,804 ops/sec ±4.94%

Fastest is [ 'arrow => {}', 'function return void 0', 'arrow => void 0', 'function return undefined', 'arrow => undefined' ]

// Run 5
arrow => void 0               x 223,014,175 ops/sec ±4.83%
arrow => undefined            x 212,708,391 ops/sec ±4.58%
arrow => {}                   x 217,071,520 ops/sec ±4.18%
function return void 0        x 216,821,677 ops/sec ±5.10%
function return undefined     x 212,494,765 ops/sec ±4.55%
function {}                   x 218,697,280 ops/sec ±4.85%

Fastest is [ 'arrow => void 0', 'function {}', 'function return void 0' ]

// Run 6
arrow => void 0               x 206,557,107 ops/sec ±5.81%
arrow => undefined            x 220,097,018 ops/sec ±4.62%
arrow => {}                   x 213,386,061 ops/sec ±5.95%
function return void 0        x 222,016,531 ops/sec ±4.91%
function return undefined     x 228,739,297 ops/sec ±4.09%
function {}                   x 222,104,228 ops/sec ±4.20%

Fastest is [ 'function return undefined', 'function {}', 'function return void 0', 'arrow => undefined', 'arrow => {}' ]

// Run 7
arrow => void 0               x 218,943,876 ops/sec ±4.87%
arrow => undefined            x 215,826,200 ops/sec ±5.00%
arrow => {}                   x 224,753,256 ops/sec ±4.61%
function return void 0        x 224,963,829 ops/sec ±5.07%
function return undefined     x 217,924,109 ops/sec ±5.27%
function {}                   x 216,230,689 ops/sec ±4.49%

Fastest is [ 'arrow => {}', 'function return void 0', 'arrow => void 0', 'function return undefined', 'function {}', 'arrow => undefined' ]

// Run 8
arrow => void 0               x 225,274,725 ops/sec ±4.56%
arrow => undefined            x 231,810,269 ops/sec ±4.14%
arrow => {}                   x 212,084,366 ops/sec ±5.10%
function return void 0        x 220,773,590 ops/sec ±5.07%
function return undefined     x 221,866,201 ops/sec ±4.86%
function {}                   x 227,474,012 ops/sec ±4.40%

Fastest is [ 'arrow => undefined', 'function {}', 'arrow => void 0', 'function return undefined', 'function return void 0' ]

// Run 9
arrow => void 0               x 213,418,000 ops/sec ±4.43%
arrow => undefined            x 220,384,510 ops/sec ±4.61%
arrow => {}                   x 224,166,715 ops/sec ±3.99%
function return void 0        x 221,609,962 ops/sec ±5.28%
function return undefined     x 219,354,272 ops/sec ±4.49%
function {}                   x 219,339,004 ops/sec ±4.66%

Fastest is [ 'arrow => {}', 'arrow => undefined', 'function return void 0', 'function return undefined', 'function {}', 'arrow => void 0' ]

// Run 10
arrow => void 0               x 214,680,508 ops/sec ±5.91%
arrow => undefined            x 222,804,892 ops/sec ±4.50%
arrow => {}                   x 213,088,462 ops/sec ±5.32%
function return void 0        x 224,056,476 ops/sec ±4.51%
function return undefined     x 214,583,384 ops/sec ±4.93%
function {}                   x 221,865,774 ops/sec ±4.94%

Fastest is [ 'function return void 0', 'arrow => undefined', 'function {}', 'function return undefined', 'arrow => void 0', 'arrow => {}' ]
```

* After running 10 benchmarks, performance differences turned out to be statistically negligible.
* So we can simply choose the version that best fits readability, intention, and linting preferences.

Does this version look better to you?

```ts
export const noop = (): void => undefined
```